### PR TITLE
Dynamic group fixes

### PR DESCRIPTION
**Pull request by rowanseymour**
* Exposes group query over the API - CasePro will use this to determine if a group is dynamic, and so not try to add/remove contacts from it.
* Prevents the contact bulk action endpoint from manually adding/removing contacts from dynamic groups
* Don't display dynamic groups in omnibox for add_group flow actions

### CODE REVIEW

**Comment by rowanseymour** [comment]
There almost certainly flows out there adding/removing contacts from dynamic groups. If we deploy this as is, those actions will error. That would certainly help track down those flows and get people to fix them, or we write a migration to remove those actions, or we have the action swallow errors.. or maybe ignore them but log something to sentry.. thoughts?

**Comment by rowanseymour** [comment]
Updated Add/Remove group flow actions to silently fail for regular contacts - but show error for test contacts

File: templates/partials/action_editor.haml [hunk 1]
```diff
@@ -45,7 +45,7 @@
       %p{class:"description"}
         -blocktrans
           Choose one or more groups below.
-      %input{ng-model:"omnibox",name:"omnibox",required:"",omnibox:'{"types":"g", "arbitraryAdd":true, "completions":true, "allowExtra": true}',groups:"action.groups",type:"hidden"}
+      %input{ng-model:"omnibox",name:"omnibox",required:"",omnibox:'{"types":"s", "arbitraryAdd":true, "completions":true, "allowExtra": true}',groups:"action.groups",type:"hidden"}
```
Comment (rowanseymour) [inline templates/partials/action_editor.haml]: g = all groups
s = static (i.e. non-dynamic) groups

**Comment by ericnewcomer** [comment]
I think it's ok to fail silently for the user in this case so long as it is reported to sentry and we notify them accordingly. We could have a flow version migration (not a db migration), but that might be equally confusing. There's also cases where action sets might vanish entirely and need to be patched. 

Alternatively, why not shell in and figure which ones do this and be proactive about this before pushing? Either way we may want to consider a flow migration for the rollback case.

**Comment by nicpottier** [comment]
What would we do even if we found the flows that were trying to modify dynamic groups? Remove the action? Or just email the people and tell them don't do that?  Seems to me maybe both of those?

In any case, this looks good and is certainly needed.

**Comment by rowanseymour** [comment]
Now with migration to check and reevaluate all dynamic groups.

**Comment by ericnewcomer** [comment]
So what about the flow migration to remove these bogus actions? Logic there doesn't seem too bad. If there's no actions left we'd route all incoming to the destination of the actionset to be removed.

It's a bit confusing for folks to visit their flows and have those just be missing, but probably better than misleading them about adding to group that doesn't happen.

Or another thought.. modify the server to fail on saving these flows with a specific error message (seems we'd want this protection anyways so the UI can't create bogus flows). This way they get feedback when they try to edit the flow and they won't be able to make changes until they correct it. Downsides are that they can revert to a bad state and also that they don't get feedback about the error until they attempt to change the flow (which is particularly unfortunate in the revert case)

**Pull request merged**